front/components/data_source/DataSourceEdition.tsx

```diff
@@ -0,0 +1,324 @@
+import {
+  Avatar,
+  Button,
+  CloudArrowLeftRightIcon,
+  Dialog,
+  Icon,
+  LockIcon,
+  Modal,
+  Page,
+} from "@dust-tt/sparkle";
+import type {
+  ConnectorProvider,
+  ConnectorType,
+  EditedByUser,
+  UpdateConnectorRequestBody,
+  UserType,
+  WhitelistableFeature,
+  WorkspaceType,
+} from "@dust-tt/types";
+import { CONNECTOR_TYPE_TO_MISMATCH_ERROR } from "@dust-tt/types";
+import { InformationCircleIcon } from "@heroicons/react/20/solid";
+import type { NextRouter } from "next/router";
+import React, { useContext, useState } from "react";
+
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import { CONNECTOR_CONFIGURATIONS } from "@app/lib/connector_providers";
+import { formatTimestampToFriendlyDate } from "@app/lib/utils";
+import { setupConnection } from "@app/pages/w/[wId]/builder/data-sources/managed";
+
+export type DataSourceIntegration = {
```

Comment:
IIRC, this type already exist in `managed.tsx`, you introduced it a few days, hours ago?

Reply:
I messed up my rebase, it's supposed to be here now.